ncm-hostsfile/src/main/pan/components/hostsfile/schema.pan

```diff
@@ -14,10 +14,21 @@ declaration template components/hostsfile/schema;
 
 include 'quattor/schema';
 
+type hostsfile_entry = {
+    "ipaddr" : type_ip
+    "aliases" ? string_trimmed
+    "comment" ? string_trimmed
+};
+
 type component_hostsfile_type = {
     include structure_component
     "file" ? string        # File to store in.  Default is /etc/hosts
-    "entries" : dict
+    "entries" : hostsfile_entry{} with {
+        foreach (k; v; SELF) {
+            if (! is_fqdn(k)) error('Invalid FQDN "%s" for hostfile entry', k);
```

Comment:
We add localhost to our entries - would that pass the is_fqdn() check?

```
+-entries
  +-localhost
  $ ipaddr : '127.0.0.1'
```

I don't think we can remove that now either because all of our localhost entries now have the # NCM suffix.

Reply:
ah ok, done